collector/perf_linux.go

```diff
@@ -49,23 +66,55 @@ func NewPerfCollector() (Collector, error) {
 		perfSwProfilers:    map[int]perf.SoftwareProfiler{},
 		perfCacheProfilers: map[int]perf.CacheProfiler{},
 	}
-	ncpus := runtime.NumCPU()
-	for i := 0; i < ncpus; i++ {
+
+	start := 0
+	ncpus := 0
+	var err error
+
+	if !isValidCPUString(cpus) {
+		start = 0
+		ncpus = runtime.NumCPU() - 1
+	} else {
+		cpu_range := strings.Split(*cpus, "-")
```

Comment:
variables in go should be camel case, so this would be cpuRange
@SuperQ Any idea why we don't catch this in CI? I assume we run go lint or something?

Reply:
Good question, we're running [golangci-lint](https://github.com/golangci/golangci-lint) in CI.